galley/pkg/server/components/processing2.go

```diff
@@ -211,12 +216,20 @@ func (p *Processing2) Start() (err error) {
 		address = p.args.APIAddress[idx+3:]
 	}
 
-	if p.listener, err = netListen(network, address); err != nil {
-		err = fmt.Errorf("unable to listen: %v", err)
-		return
+	mcp.RegisterResourceSourceServer(p.grpcServer, p.mcpSource)
+	if p.args.SecureGRPC != nil {
+		mcp.RegisterResourceSourceServer(p.args.SecureGRPC, p.mcpSource)
 	}
 
-	mcp.RegisterResourceSourceServer(p.grpcServer, p.mcpSource)
+	if p.args.InsecureGRPC == nil {
+		if p.listener, err = netListen(network, address); err != nil {
```

Comment:
Move lines 208-217 into the conditional. We don't need to compute the network address if the server is already provided for us.

Reply:
Done